packages/frames-validator/src/validation.ts

```diff
@@ -10,51 +13,80 @@ export type * from "./types.js";
 
 const { b64Decode } = fetcher;
 
-export function validateFramesPost(
+export async function validateFramesPost(
   data: XmtpOpenFramesRequest,
-): XmtpValidationResponse {
+  env?: XmtpEnv,
+): Promise<XmtpValidationResponse> {
   const { untrustedData, trustedData } = data;
   const { walletAddress } = untrustedData;
   const { messageBytes: messageBytesString } = trustedData;
 
   const messageBytes = b64Decode(messageBytesString);
 
-  const { actionBody, actionBodyBytes, signature, signedPublicKeyBundle } =
-    deserializeProtoMessage(messageBytes);
-
-  const verifiedWalletAddress = getVerifiedWalletAddress(
+  const {
+    actionBody,
     actionBodyBytes,
     signature,
     signedPublicKeyBundle,
-  );
+    installationId, // not necessary
+    installationSignature,
+    inboxId,
+  } = deserializeProtoMessage(messageBytes);
+
+  const isV2Frame = signature && signedPublicKeyBundle;
+
+  if (isV2Frame) {
+    const verifiedWalletAddress = getVerifiedWalletAddress(
+      actionBodyBytes,
+      signature,
+      signedPublicKeyBundle,
+    );
 
-  if (verifiedWalletAddress !== walletAddress) {
-    console.log(`${verifiedWalletAddress} !== ${walletAddress}`);
-    throw new Error("Invalid wallet address");
+    if (verifiedWalletAddress !== walletAddress) {
+      console.log(`${verifiedWalletAddress} !== ${walletAddress}`);
+      throw new Error("Invalid wallet address");
+    }
+  } else {
+    // make sure inbox IDs match
+    const addressInboxId = await getInboxIdForAddress(walletAddress, env);
```

Comment:
The wallet address is untrusted data. You can make it say anything. 

Having it match the inbox ID doesn't actually tell us that the signer is a member of an inbox with that wallet address.